lib/mongo/socket.rb

```diff
@@ -169,11 +169,27 @@ def eof?
     private
 
     def read_from_socket(length)
-      data = String.new
+      data = ('x'*length).force_encoding('BINARY')
       deadline = (Time.now + timeout) if timeout
+      # OpenSSL reads in 16 KB chunks max, there is no benefit to allocating
+      # a larger buffer as extra space won't be used anyway:
+      # https://linux.die.net/man/3/ssl_read
```

Comment:
This comment can move to `ssl.rb`.

Reply:
Yep, will reword it to be more relevant to the way the code is now.